[Doc] Revamp ray core design patterns doc [17/n]: tree of actors (#30188)
ray-project/ray: doc/source/ray-core/patterns/tree-of-actors.rst

Pattern: Using a supervisor actor to manage a tree of actors
============================================================

> In this pattern, a collection of Ray worker actors is managed by a supervisory Ray actor.

Reviewer: ```suggestion
In this pattern, a collection of Ray worker actors is managed by a supervising Ray actor.
```

> In this pattern, a collection of Ray worker actors is managed by a supervisory Ray actor.

Reviewer: I think we need to start with a description of why you would want to do this. Currently it is not clear what the benefit is.

Pattern: Using a supervisor actor to manage a tree of actors
============================================================

> Actor supervision is a pattern that a supervising Ray actor manages a collection of Ray worker actors.

Reviewer: ```suggestion
Actor supervision is a pattern in which a supervising actor manages a collection of worker actors.
```

The supervisor delegates tasks to subordinates and handles their failures.

> This pattern simplifies the driver since it only needs to manage a few supervisors instead of a lot of worker actors
> and doesn't need to deal with failures from worker actors directly.

Reviewer: ```suggestion
This pattern simplifies the driver since it manages only a few supervisors and does not deal with failures from worker actors directly.
```
